### PR TITLE
[PUB-2129] Fix styling and close behavior for user tags modal

### PR DESCRIPTION
**Pull request by hamstu**
## Description

The changes are as mentioned in the title, there's a bunch of ESLint / Prettier changes as well, so be sure to use 'No Whitespace' in the diff view, which will help a bit (still fairly messy diffs, sorry! 😅)

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx [hunk 1]
```diff
@@ -1,113 +1,153 @@
 import React from 'react';
```
Comment (hamstu) [inline packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx]: Most of the changes to this file are Prettier / ESLint. Make sure you have 'No Whitespace' enabled in Git to help a bit.

File: packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx [hunk 2]
```diff
@@ -1,113 +1,153 @@
...
-  close = (ev) => {
-    var node = ev.target;
+  close = ev => {
+    const node = ev.target;
 
     if (node.id === 'overlay') {
+      const { hideCloseButton } = this.props;
+      if (hideCloseButton) {
+        return;
+      }
```
Comment (hamstu) [inline packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx]: **Actual change here! ** I added this code to make sure that the same way we prevent closing when you press Esc and hideCloseButton is true, we do the same for clicking the overlay.

File: packages/composer/composer/components/css/InstagramUserTags.css [hunk 3]
```diff
@@ -1,62 +1,3 @@
...
-  color: #2c4bff;
-}
-
-.modalStyles .atSign {
-  font-size: 20px;
-  align-items: center;
-  display: inline-flex;
-  margin-right: 6px;
+.modal {
+  width: auto;
```
Comment (hamstu) [inline packages/composer/composer/components/css/InstagramUserTags.css]: Ensure the modal is centered correctly.

File: packages/composer-user-tags/components/UserTags/index.jsx [hunk 4]
```diff
@@ -28,6 +28,9 @@ const calculateTagStyles = ({ tag, showTags }) => ({
   display: showTags ? 'block' : 'none',
 });
 
+/** React is most efficient with unique keys for items. Quick method to create them based on a given string. */
+const uniqKey = identifier => `${identifier}-${new Date().getTime()}`;
```
Comment (albennett) [inline packages/composer-user-tags/components/UserTags/index.jsx]: I think I've seen Ana use the `uuid` package to get a unique identifier.

File: packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx (hunk 1 above)
Comment (albennett) [reply packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx]: Thanks! Gosh, that package name 😅

File: packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx (hunk 2 above)
Comment (albennett) [reply packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx]: Hm, I don't know if there's a way to close out of the ig video thumbnail editor with this change 🤔 May want to check all the areas where we hide the close button.

File: packages/composer-user-tags/components/UserTags/index.jsx [hunk 5]
```diff
@@ -94,12 +100,10 @@ const UserTags = ({ media, userTags = [], saveGlobalTags }) => {
...
       y: PropTypes.string,
     }).isRequired,
-    index: PropTypes.number.isRequired,
   };
 
-  // TO-DO: refactor to be more a11y friendly
-  const TagListItem = ({ tag, index }) => (
-    <UserName key={index}>
+  const TagListItem = ({ tag }) => (
+    <UserName key={uniqKey(tag.username)}>
```
Comment (albennett) [inline packages/composer-user-tags/components/UserTags/index.jsx]: Thanks for replacing these with better unique keys! 💯

File: packages/composer-user-tags/components/UserTags/index.jsx (hunk 4 above)
Comment (hamstu) [reply packages/composer-user-tags/components/UserTags/index.jsx]: Good one! I hadn't thought to check for that. Might switch to using it for consistency! But I know you might be touching this part of the code too, so let me know.

File: packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx (hunk 2 above)
Comment (hamstu) [reply packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx]: Good one! I'm glad you brought it up then. I'll validate that before I merge this in. If it's breaking I think I'll just add another prop called `preventOverlayClose` or something like that.

File: packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx (hunk 1 above)
Comment (hamstu) [reply packages/composer/composer/__legacy-buffer-web-shared-components__/modal/modal.jsx]: Ugh right!? Let's change it. We don't care about `buffer-web` anymore! (Right?)

File: packages/composer-user-tags/components/UserTags/index.jsx (hunk 4 above)
Comment (albennett) [reply packages/composer-user-tags/components/UserTags/index.jsx]: Awesome! I'll use the uniqKey logic (may just need to fix a merge conflict)

**Comment by albennett** [comment]
Just checked out your new changes. Thanks for renaming & moving the modal component. After you manually test, LGTM to deploy! 🚀 

Side note: Clicking `no whitespace` doesn't seem to cover most of the prettier changes. Could be a good one to think through with our prettier approach, as PRs will be hard to read.

**Comment by hamstu** [comment]
Yeah, great one re: Prettier. It's noisy, and since it's automatic with my editor it sort of just happened 😅 . This is a good learning, so my plan for next time is to turn off format on save and do it all in one commit at the end. At least that way you can filter out the commmit with GitHub.

image

**Pull request merged**